R/preparedata.R

```diff
@@ -1083,7 +1083,7 @@ processInputData <- function(x, subset, weights)
     # Try to use S3 method to extract data
     x <- ExtractChartData(x)
     n.dim <- length(dim(x)) - isQTableWithMultStatistic(x)
-    if (n.dim > 2)
+    if (n.dim >= 2)
```

Comment:
What does it mean to flatten a 2D table? To me it seems a 2D table doesn't need flattening?

Reply:
That's what I thought too. But it turns out the flattening is also removing the spans and adding it to the row/column labels: https://displayr.slack.com/archives/C013HGKCK9V/p1748868145503909

Reply:
So with 1D tables we remove the spans and add it to the row labels?

Reply:
No, but it didn't do that before, so we don't want to change any existing behaviour (e.g, the 1-d chart in the original bug report would change if we tried to make this consistent).

Reply:
I think when we change to tidy tables, we should deal with this consistently, but for now I think we just try not to break anything that has already been created.